GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -748,8 +748,8 @@ private void LoadUnstagedOutput(IList<GitItemStatus> allChangedFiles)
                 else
                     unStagedFiles.Add(fileStatus);
             }
-            Unstaged.GitItemStatuses = unStagedFiles;
-            Staged.GitItemStatuses = stagedFiles;
+            Unstaged.SetDiffs(new GitRevision(GitRevision.UnstagedGuid), new GitRevision(GitRevision.IndexGuid), unStagedFiles);
+            Staged.SetDiffs(new GitRevision(GitRevision.IndexGuid), new GitRevision("HEAD"), stagedFiles);
```

Comment:
I think we can created static instances for these:
* `new GitRevision(GitRevision.UnstagedGuid)`
* `new GitRevision(GitRevision.IndexGuid)`
* `new GitRevision("HEAD")` (can we?)

This way we can simplify the API.

I've been working on getting all behaviors out of GitRevision and having it as an immutable POCO. So in the target state, once initialised they can't be changed.

Something like the following maybe?

```
    public sealed class GitRevision : IGitItem, INotifyPropertyChanged
    {
        public static readonly GitRevision EmptyRevision = new GitRevision("");
        public static readonly GitRevision HeadRevision = new GitRevision("HEAD");
        public static readonly GitRevision IndexRevision = new GitRevision(IndexGuid);
        public static readonly GitRevision UnstagedRevision = new GitRevision(UnstagedGuid);

```

Reply:
With #4556 we probably should get the GitRevision from RevisionGrid and not create new objects.
(In the current version of #4556, new GitRevision("HEAD") is not permitted.)

Reply:
/cc: @drewnoakes thoughts?

Reply:
RevisionGrid is not available for FormCommit right now, will not change now.

Reply:
With respect to #4556 restricting GitRevision to ObjectId, it seems like from @gerhardol's comments there that such a change is too restrictive, and I defer to the judgement of those with more experience. My intent is to introduce ObjectId only where it is completely non-controversial to do so, and then see whether it adds any value overall.

Reply:
I like the idea in general, I think it is a move in a right direction